test/vfd.c

```diff
@@ -3420,6 +3474,64 @@ test_splitter(void)
 
 #undef SPLITTER_TEST_FAULT
 
+/*****************************************************************************
+ *
+ * Function    setup_rand()
+ *
+ * Purpose:    Use gettimeofday() to obtain a seed for rand(), print the
+ *             seed to stdout, and then pass it to srand().
+ *
+ *             This is a version of the same routine in
+ *             testpar/t_cache.c modified for use in serial tests.
+ *
+ * Return:     void.
+ *
+ * Programmer: JRM -- 6/20/20
+ *
+ * Modifications:
+ *
+ *             None.
+ *
+ *****************************************************************************/
+static void
+setup_rand(void)
+{
+    hbool_t        use_predefined_seed = FALSE;
+    unsigned       predefined_seed     = 18669;
+    unsigned       seed;
+    struct timeval tv;
+
+    if (use_predefined_seed) {
+
+        seed = predefined_seed;
+
+        HDfprintf(stdout, "\n%s: predefined_seed = %d.\n\n", __func__, seed);
+        HDfflush(stdout);
+
+        HDsrand(seed);
+    }
+    else {
+
+        if (HDgettimeofday(&tv, NULL) != 0) {
+
+            HDfprintf(stdout, "\n%s: gettimeofday() failed -- srand() not called.\n\n", __func__);
+            HDfflush(stdout);
+        }
+        else {
```

Comment:
The else clause is silly. The only way gettimeofday can fail is if the struct timeval pointer is bad, which seems unlikely as it's on the stack.

Reply:
I think since gettimeofday() is capable of returning an error according to the man page (is it in a standard?), we should handle an error correctly here even if the current version of linux might never fail.